### PR TITLE
Added changes in business financials - Discount and Margin, Payout and Taxes tabs

### PR DESCRIPTION
**Pull request by saicharan-reddy1**
-Added discount margin tab changes (restricting tab based upon account status)
-Incorporated additional changes to payout and taxes tab (void cheque, fields modification)
image
image
image
image

### CODE REVIEW

File: components/DiscountAndMargin/DiscountAndMarginTab.tsx [hunk 1]
```diff
@@ -49,12 +53,12 @@ export function DiscountAndMarginTab() {
   const discountMarginPercentage =
     data?.practitioner?.map(
       (index) => index?.dispensary?.discountPercentage
-    )[0] || null;
+    )[0] || 0;
```
Comment (joneskebhart9787) [inline components/DiscountAndMargin/DiscountAndMarginTab.tsx:56]: Are we getting discount margin percentage in an array?

File: components/DiscountAndMargin/PayoutAndTaxTab.tsx [hunk 2]
```diff
@@ -23,89 +23,130 @@ import {  useForm } from "react-hook-form";
...
 import { NotificationProps } from "../Notifications/Notifications";
-import { AccountHeldBy, AccountType, BusinessType, DispensaryPayoutTaxUpdateInput, PayoutAndTaxesDetailsQuery, useDispensaryPayoutTaxUpdateMutation, usePayoutAndTaxesDetailsLazyQuery, usePayoutAndTaxesDetailsQuery } from "@/saleor/api";
+import { AccountType, BusinessType, ProfitAccountCreateMutationVariables, usePayoutAndTaxesDetailsQuery, useProfitAccountCreateMutation, usePayoutAndTaxesPersonalInfoQuery } from "@/saleor/api";
 import useProvinceData from "@/lib/hooks/useProvinceData";
 import translations,{messages} from "@/components/translations";
-import { ACCOUNT_TYPE_OPTIONS, BUSINESS_OPTIONS, HELD_BY_OPTIONS } from "@/components/DropdownOptions";
+import { ACCOUNT_TYPE_OPTIONS, BUSINESS_OPTIONS, SHOW_GST_NUM_OPTIONS } from "@/components/DropdownOptions";
+import { FileUploader } from "../FormComponents/FileUpload/FilesUploader";
+import { uploadFileWithMutation } from "../utils/uploadFileWithMutation";
+import { CHEQUE, FILE, POST_METHOD, PROFIT_MARGIN, VARIABLE } from "@/lib/const";
```
Comment (joneskebhart9787) [inline components/DiscountAndMargin/PayoutAndTaxTab.tsx:32]: Please remove unused imports

File: components/DiscountAndMargin/PayoutAndTaxTab.tsx [hunk 3]
```diff
@@ -23,89 +23,130 @@ import {  useForm } from "react-hook-form";
...
-      return;
-    }
-    reset(getDefaultFormValues(data));
-  }, [data]);
-  const [dispensaryPayoutTaxUpdate] = useDispensaryPayoutTaxUpdateMutation();
-  const submitHandler = async (formData: DispensaryPayoutTaxUpdateInput) => {
+    reset(getDefaultFormValues());
+  }, [defaultPersonalData]);
+
+  const [profitAccountCreateMutation] = useProfitAccountCreateMutation();
```
Comment (joneskebhart9787) [inline components/DiscountAndMargin/PayoutAndTaxTab.tsx:104]: Where are you using this mutation?

File: components/Dispensary/dispensarySetting.tsx [hunk 4]
```diff
@@ -10,16 +10,21 @@ import Link from "next/link";
 import { usePaths } from "@/lib/paths";
 import { messages } from "@/components/translations";
 import { useLocalStorage } from "@/lib/hooks/useLocalStorage";
-import { USER_DETAILS } from "@/lib/const";
+import { EMPTY, USER_DETAILS } from "@/lib/const";
+import { useAccessDetailsQuery, useAccountStatusQuery } from "@/saleor/api";
```
Comment (joneskebhart9787) [inline components/Dispensary/dispensarySetting.tsx:14]: Please check whether you are using useAccessDetailsQuery?

File: components/DiscountAndMargin/DiscountAndMarginTab.tsx (hunk 1 above)
Comment (saicharan-reddy1) [reply components/DiscountAndMargin/DiscountAndMarginTab.tsx:56]: Yes, in practitioner array.

File: components/DiscountAndMargin/PayoutAndTaxTab.tsx (hunk 2 above)
Comment (saicharan-reddy1) [reply components/DiscountAndMargin/PayoutAndTaxTab.tsx]: Removed

File: components/Dispensary/dispensarySetting.tsx (hunk 4 above)
Comment (saicharan-reddy1) [reply components/Dispensary/dispensarySetting.tsx]: I have removed unused imports.

File: components/DiscountAndMargin/PayoutAndTaxTab.tsx (hunk 3 above)
Comment (saicharan-reddy1) [reply components/DiscountAndMargin/PayoutAndTaxTab.tsx]: Removed as we are not using it.

**Review by joneskebhart9787 — DISMISSED** [review]

File: components/FormComponents/FileUpload/FilesUploader.tsx [hunk 5]
```diff
@@ -71,73 +77,50 @@ export const FileUploader = ({
...
-                <NavIconButton className="w-4 h-4" icon="close" />
-              </button>
+        {
+          selectedFiles.length > 0 && 
+          <>
+            Selected files:
+            <div className="w-full justify-between mt-2">
+              <ul className="grid grid-col-1 sm:grid-cols-2 w-full gap-1">
+                {selectedFiles.map((file, index) => (
+                  // eslint-disable-next-line react/no-array-index-key
```
[the remark this answers is not in the document]
Comment (saicharan-reddy1) [reply components/FormComponents/FileUpload/FilesUploader.tsx:101]: This code is already present. I have added conditional check above this.

File: components/FormComponents/FileUpload/FilesUploader.tsx [hunk 6]
```diff
@@ -71,73 +77,50 @@ export const FileUploader = ({
...
+                  <div
+                    className="flex border-gray-600 border h-8 items-center flex justify-between w-full p-2 font-semibold overflow-hidden gap-1"
+                    // eslint-disable-next-line react/no-array-index-key
```
[the remark this answers is not in the document]
Comment (saicharan-reddy1) [reply components/FormComponents/FileUpload/FilesUploader.tsx:104]: Same as above.

**Review by joneskebhart9787 — APPROVED** [review]

**Review by visak-kumar — APPROVED** [review]

**Pull request merged**